### PR TITLE
feat: add k8 abbreviates in jump to kind under resource browser

### PR DESCRIPTION
**Pull request by Elessar1802**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Additionally fixed a bug that would show `cluster overview` as a option in jump to kind which should not be the case.

Fixes https://github.com/devtron-labs/devtron/issues/4739

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR require documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code
* [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/components/ResourceBrowser/Types.ts [hunk 1]
```diff
@@ -217,3 +217,11 @@ export interface K8sObjectOptionType extends OptionType {
     }
     groupName: string
 }
+
+export interface K8Abbreviates {
+    [key: string]: string,
+}
+
+export interface K8AbbreviatesResponse extends ResponseType {
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/Types.ts:225]: ResponseType<K8Abbreviates>

File: src/components/ResourceBrowser/Types.ts [hunk 2]
```diff
@@ -217,3 +217,11 @@ export interface K8sObjectOptionType extends OptionType {
     }
     groupName: string
 }
+
+export interface K8Abbreviates {
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/Types.ts:221]: format

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx [hunk 3]
```diff
@@ -62,6 +65,16 @@ const Sidebar = ({
         }
     }, [k8SObjectMap?.size, sideBarElementRef.current])
 
+    useEffect(() => {
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/Sidebar.tsx:68]: we can use useAsync

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx [hunk 4]
```diff
@@ -272,10 +272,29 @@ const Sidebar = ({
     }
 
     const handleInputChange = (newValue: string, actionMeta: InputActionMeta): void => {
-        if (actionMeta.action === 'input-change') {
-            setSearchText(newValue)
-            setMenuOpen(!!newValue)
+        if (actionMeta.action !== 'input-change') {
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/Sidebar.tsx:275]: Check can have a constant if possible

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx [hunk 5]
```diff
@@ -272,10 +272,29 @@ const Sidebar = ({
...
+            return
         }
+
+        const loc = k8sObjectOptionsList.findIndex((value) => value.label.toLowerCase() === abbreviate_expanded)
+        if (loc === -1) {
+            return
+        }
+
+        /* swap element at @loc with element at position 0 */
+        const temp = k8sObjectOptionsList[0]
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/Sidebar.tsx:293]: can we create a swap method in Helpers if not already present.

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx (hunk 4 above)
Comment (Elessar1802) [reply src/components/ResourceBrowser/ResourceList/Sidebar.tsx:191]: TODO: will be incorporated from the feat/user-status-p3 branch

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx [hunk 6]
```diff
@@ -34,11 +38,20 @@ const Sidebar = ({
...
-    const [k8sObjectOptionsList, setK8sObjectOptionsList] = useState<K8sObjectOptionType[]>([])
+    const [, k8Abbreviates] = useAsync(getK8Abbreviates)
     const sideBarElementRef = useRef<HTMLDivElement>(null)
     const preventScrollRef = useRef<boolean>(false)
     const searchInputRef = useRef<Select<K8sObjectOptionType, false, GroupBase<K8sObjectOptionType>>>(null)
+    const k8sObjectOptionsList = useMemo(() => convertK8sObjectMapToOptionsList(k8SObjectMap), [k8SObjectMap?.size])
+
+    const handleInputShortcut = (e: React.KeyboardEvent<HTMLDivElement>) => {
+        const _key = e.key
+        if (_key === 'k') {
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/Sidebar.tsx:49]: try not to use else if

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx [hunk 7]
```diff
@@ -272,14 +187,32 @@ const Sidebar = ({
...
+        // TODO: replace with enum after merge of feat/user-status-p3
+        if (actionMeta.action !== 'input-change') {
+            return
+        }
+
+        setSearchText(newValue)
+    }
+
+    const bringMatchedAbbreviatedOptionToFront = (options: K8sObjectOptionType[]): K8sObjectOptionType[] => {
+        if (!searchText || !k8Abbreviates) {
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/Sidebar.tsx:199]: Can we clean the conditions by moving searchText?.toLowerCase() above and encapsulate final conditions into better names

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx [hunk 8]
```diff
@@ -272,14 +187,32 @@ const Sidebar = ({
...
+        }
+        const lowerSearchText = searchText.toLowerCase()
+        if (!k8Abbreviates[lowerSearchText]) {
+            return options
         }
+        const loc = k8sObjectOptionsList.findIndex(
+            (option) => k8Abbreviates[lowerSearchText] === option.label.toLowerCase(),
+        )
+        if (loc > -1) {
+            swap(options, loc, 0)
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/Sidebar.tsx:210]: Confirm this with @vivek-devtron

File: src/components/ResourceBrowser/ResourceList/Sidebar.tsx [hunk 9]
```diff
@@ -296,33 +229,38 @@ const Sidebar = ({
...
+                    {!formatOptionLabelMeta.inputValue ? (
+                        <span className="w-100 dc__ellipsis-right">{option.label}</span>
+                    ) : (
+                        <span
+                            className="w-100 dc__ellipsis-right"
+                            dangerouslySetInnerHTML={{
+                                __html: option.label.replace(
+                                    new RegExp(formatOptionLabelMeta.inputValue, 'gi'),
+                                    (highlighted) => `<mark>${highlighted}</mark>`,
+                                ),
```
Comment (AbhishekA1509) [inline src/components/ResourceBrowser/ResourceList/Sidebar.tsx:249]: Check if we can use highlightSearchText from fe-common and use DomPurify whenever using dangerouslySetInnerHTML

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**